optd-cost-model/src/cost_model.rs

```diff
@@ -114,7 +201,17 @@ impl<S: CostModelStorageManager + Send + Sync + 'static> CostModel for CostModel
             PhysicalNodeType::PhysicalSort | PhysicalNodeType::PhysicalProjection => {
                 Ok(children_statistics[0].clone())
             }
-        }
+        }?;
+        if store_output_statistic {
+            let res = self
+                .storage_manager
+                .store_cost(context.expr_id, None, Some(output_statistic.clone()), None)
+                .await;
+            if res.is_err() {
+                eprintln!("Failed to store output statistic");
+            }
+        };
+        Ok(output_statistic)
```

Comment:
Sure, I think the retry should be handled by the storage manager, and I will add a comment there. I don't think it is a big deal, since maybe we can recalculate it again?